test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -654,6 +529,126 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", Ordered, func() {
+
+			// It's possible that when this test runs the value of
+			// defaultCpuNotInSchedulingDomains is empty if no gu pods are running
+			defaultCpuNotInSchedulingDomains, err := getCPUswithLoadBalanceDisabled(ctx, workerRTNode)
+			Expect(err).ToNot(HaveOccurred(), "Unable to fetch scheduling domains")
+			if len(defaultCpuNotInSchedulingDomains) > 0 {
+				pods, err := pods.GetPodsOnNode(context.TODO(), workerRTNode.Name)
+				if err != nil {
+					testlog.Warningf("cannot list pods on %q: %v", workerRTNode.Name, err)
+				} else {
+					testlog.Infof("pods on %q BEGIN", workerRTNode.Name)
+					for _, pod := range pods {
+						testlog.Infof("- %s/%s %s", pod.Namespace, pod.Name, pod.UID)
+					}
+					testlog.Infof("pods on %q END", workerRTNode.Name)
+				}
+				Expect(defaultCpuNotInSchedulingDomains).To(BeEmpty(), "the test expects all CPUs within a scheduling domain when starting")
+			}
+
+			By("Starting the pod")
+			testpod.Spec.NodeSelector = testutils.NodeSelectorLabels
+			testpod.Spec.Containers[0].Image = "quay.io/mniranja/busycpus"
+			testpod.Spec.Containers[0].Resources.Limits[corev1.ResourceCPU] = resource.MustParse("2")
+			err = testclient.Client.Create(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+			testpod, err = pods.WaitForCondition(ctx, client.ObjectKeyFromObject(testpod), corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+			logEventsForPod(testpod)
+			Expect(err).ToNot(HaveOccurred(), "failed to create guaranteed pod %v", testpod)
+			allTestpods[testpod.UID] = testpod
+		})
+
+		AfterAll(func() {
+			for podUID, testpod := range allTestpods {
+				testlog.Infof("deleting test pod %s/%s UID=%q", testpod.Namespace, testpod.Name, podUID)
+				err := testclient.Client.Get(ctx, client.ObjectKeyFromObject(testpod), testpod)
+				Expect(err).ToNot(HaveOccurred())
+				//testpodUID := testpod.UID
+				err = testclient.Client.Delete(ctx, testpod)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = pods.WaitForDeletion(ctx, testpod, pods.DefaultDeletionTimeout*time.Second)
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+
+		Describe("cpuset controller", func() {
+			It("Verify cpuset controller interface files have right values", func() {
+				cpusetCfg := &controller.CpuSet{}
+				err := getter.Container(ctx, testpod, testpod.Spec.Containers[0].Name, cpusetCfg)
+				Expect(err).ToNot(HaveOccurred())
+				// Get cpus used by the container
+				tasksetcmd := []string{"taskset", "-pc", "1"}
+				testpodAffinity, err := pods.ExecCommandOnPod(testclient.K8sClient, testpod, testpod.Spec.Containers[0].Name, tasksetcmd)
+				podCpusStr := string(testpodAffinity)
+				parts := strings.Split(strings.TrimSpace(podCpusStr), ":")
+				testpodCpus := strings.TrimSpace(parts[1])
+				testlog.Infof("%v pod is using %v cpus", testpod.Name, string(testpodCpus))
+				Expect(cpusetCfg.Cpus).To(Equal(testpodCpus), "cpuset.cpus not matching the process affinity")
```

Comment:
The comparison here is done between strings but they can be logically equals while unequal by string.
Example:
setA: `1-3,7`
setB: `1,2,3,7`
logically equals but the strings are different.
parse them into `cpuset` and then compare.

Reply:
fixed in the latest commit